### PR TITLE
more robust previous_state_hash parsing

### PR DESCRIPTION
**Pull request by n1tranquilla**
## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have verified new and existing tests pass locally with my changes.
- [x] I verified whether it was necessary to increment the database version.

### CODE REVIEW

**Comment by n1tranquilla** [comment]
The database has already been incremented in previous PR. If tier3 does not rebuild the database, we'll have to incr the version.

File: rust/src/block/previous_state_hash.rs [hunk 1]
```diff
@@ -23,31 +22,25 @@ pub struct PreviousStateHash(pub String);
...
+
+        // Locate "previous_state_hash" within the buffer
+        let prev_state_hash_key = "\"previous_state_hash\"";
+        if let Some(hash_pos) = buffer.find(prev_state_hash_key) {
+            let hash_start = hash_pos + prev_state_hash_key.len();
+            // Find the first quote after the colon
+            if let Some(quote_start) = buffer[hash_start..].find('"') {
+                let actual_start = hash_start + quote_start + 1; // Move to the start of the hash
+                                                                 // Find the ending quote
+                if let Some(quote_end) = buffer[actual_start..].find('"') {
```
Comment (trevorbernard) [inline rust/src/block/previous_state_hash.rs:41]: The hash is fixed size so you shouldn't need to find the closing double quote

Comment (trevorbernard) [reply rust/src/block/previous_state_hash.rs:41]: I believe it's 52 characters wide

Comment (jhult) [reply rust/src/block/previous_state_hash.rs:41]: Will try to address this in a follow-up PR.

Comment (jhult) [reply rust/src/block/previous_state_hash.rs:41]: Fixed in this commit: https://github.com/Granola-Team/mina-indexer/commit/931f86f52bb55891aba064afbf302bdefc8d8d2f

**Pull request merged**